packages/blockchain-wallet-v4-frontend/src/store/index.js

```diff
@@ -24,59 +24,49 @@ import {
 } from '../middleware'
 
 const configuredStore = async function () {
-  let res
+  // immediately load app configuration
   let options
-  let assetsRes
-  let assets
-  let erc20Res
-  let erc20s
-
   try {
-    res = await fetch('/wallet-options-v4.json')
+    let res = await fetch('/wallet-options-v4.json')
     options = await res.json()
   } catch (error) {
     throw new Error('wallet-options failed to load.')
   }
-  try {
-    assetsRes = await fetch(`${options.domains.api}/assets/currencies/custodial`)
-    assets = await assetsRes.json()
-    if (!assets.currencies) throw new Error()
-  } catch (error) {
-    throw new Error('custodial currencies failed to load.')
-  }
-  try {
-    erc20Res = await fetch(`${options.domains.api}/assets/currencies/erc20`)
-    erc20s = await erc20Res.json()
-    if (!erc20s.currencies) throw new Error()
-  } catch (error) {
-    throw new Error('erc20 currencies failed to load.')
-  }
-
-  let supportedCoins = assets.currencies
-  let supportedErc20s = erc20s.currencies
 
-  const supported = supportedCoins.concat(supportedErc20s).filter((coin) => !coin.symbol.includes('.'))
+  // define empty window coins object
+  // result of web worker will populate data later
   window.coins = {}
```

Comment:
should we make it null?

maybe not, if its empty object it will throw for any coin anyway